orca-web/src/test/groovy/com/netflix/spinnaker/orca/controllers/CapabilitiesControllerTest.java

```diff
@@ -0,0 +1,78 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.controllers;
+
+import static java.lang.String.format;
+import static org.mockito.Mockito.when;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.MOCK;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.netflix.spinnaker.config.DeploymentMonitorDefinition;
+import com.netflix.spinnaker.config.MonitoredDeployConfigurationProperties;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(
+    classes = {CapabilitiesController.class},
+    webEnvironment = MOCK)
+@AutoConfigureMockMvc
+@EnableWebMvc
+@WithMockUser("Bea O'Problem")
+class CapabilitiesControllerTest {
+  @Autowired MockMvc mvc;
+
+  @MockBean MonitoredDeployConfigurationProperties configurationProperties;
+
+  private final MockHttpServletRequestBuilder request = get("/capabilities/deploymentMonitors");
+
+  @Test
+  void returnsEmptyWhenMonitoredDeployIsDisabled() throws Exception {
+    configurationProperties = null;
+
+    mvc.perform(request).andExpect(status().is2xxSuccessful()).andExpect(content().json("[]"));
+  }
+
+  @Test
+  void returnsValidData() throws Exception {
+    List<DeploymentMonitorDefinition> monitorDefinitions = new ArrayList<>();
+    DeploymentMonitorDefinition def1 = new DeploymentMonitorDefinition();
+    def1.setName("the name");
+    def1.setId("the id");
+    monitorDefinitions.add(def1);
+
+    when(configurationProperties.getDeploymentMonitors()).thenReturn(monitorDefinitions);
+
+    mvc.perform(request)
+        .andExpect(status().is2xxSuccessful())
+        .andExpect(
+            content().json(format("[{'id':'%s','name':'%s'}]", def1.getId(), def1.getName())));
```

Comment:
there has got to be a better way than `format`, feedback appreciated!

Reply:
in pure Java I don't think so. But it seems like a job for `def1`'s `toString()`